imagesize.py

```diff
@@ -55,27 +54,27 @@ def _convertToDPI(density, unit):
 
 
 def _convertToPx(value):
-    matched = re.match(r"(\d+)(?:\.\d)?([a-z]*)$", value)
+    matched = re.match(r"(\d+(?:\.\d+)?)?([a-z]*)$", value)
     if not matched:
         raise ValueError("unknown length value: %s" % value)
-    else:
-        length, unit = matched.groups()
-        if unit == "":
-            return int(length)
-        elif unit == "cm":
-            return int(length) * 96 / 2.54
-        elif unit == "mm":
-            return int(length) * 96 / 2.54 / 10
-        elif unit == "in":
-            return int(length) * 96
-        elif unit == "pc":
-            return int(length) * 96 / 6
-        elif unit == "pt":
-            return int(length) * 96 / 6
-        elif unit == "px":
-            return int(length)
-        else:
-            raise ValueError("unknown unit type: %s" % unit)
+
+    length, unit = matched.groups()
+    if unit == "":
+        return float(length)
+    elif unit == "cm":
+        return float(length) * 96 / 2.54
+    elif unit == "mm":
+        return float(length) * 96 / 2.54 / 10
+    elif unit == "in":
+        return float(length) * 96
+    elif unit == "pc":
+        return float(length) * 96 / 6
+    elif unit == "pt":
+        return float(length) * 96 / 6
+    elif unit == "px":
```

Comment:
Do all values really need to be parsed as `float`?

Reply:
I think we should do this because we look for it in the above regular expression.
Also the height or width could always be a ``float``, why not parse it as a ``float``?

Reply:
Float just doesn't seem right to me regarding unit `px`, but probably you are right.